lib/paloaltofw.py

```diff
@@ -0,0 +1,583 @@
+    _PLATFORM = 'paloalto'
+    SUFFIX = '.xml'
+    _SUPPORTED_AF = set(('inet', 'inet6', 'mixed'))
+    _AF_MAP = {'inet': (4,),
+               'inet6': (6,),
+               'mixed': (4, 6)}
+
+    INDENT = '  '
+
+    def _BuildTokens(self):
+        """Build supported tokens for platform.
+
+        Returns:
+          tuple containing both supported tokens and sub tokens
+        """
+        supported_tokens, supported_sub_tokens = super(
+            PaloAltoFW, self)._BuildTokens()
+
+        supported_tokens |= {'expiration',
+                             'logging',
+                             'pan_application',
+                             'owner',
+                             'timeout'}
+
+        supported_sub_tokens.update(
+            {'action': {'accept', 'deny', 'reject', 'count', 'log'},
+             })
+        del supported_sub_tokens['option']
+        return supported_tokens, supported_sub_tokens
+
+    def _TranslatePolicy(self, pol, exp_info):
+        """Transform a policy object into a PaloAltoFW object.
+
+        Args:
+          pol: policy.Policy object
+          exp_info: print a info message when a term is set to expire
+                    in that many weeks
+
+        Raises:
+          UnsupportedFilterError: An unsupported filter was specified
+          UnsupportedHeader: A header option exists that is not understood/usable
+          PaloAltoFWDuplicateTermError: Two terms were found with same name in same filter
+        """
+        self.pafw_policies = []
+        self.addressbook = collections.OrderedDict()
+        self.applications = []
+        self.pan_applications = []
+        self.ports = []
+        self.from_zone = ''
+        self.to_zone = ''
+        self.policy_name = ''
+
+        current_date = datetime.date.today()
+        exp_info_date = current_date + datetime.timedelta(weeks=exp_info)
+        for header, terms in pol.filters:
+            if self._PLATFORM not in header.platforms:
+                continue
+
+            filter_options = header.FilterOptions(self._PLATFORM)
+
+            if (len(filter_options) < 4 or filter_options[0] != 'from-zone' or
+                    filter_options[2] != 'to-zone'):
+                raise UnsupportedFilterError(
+                    'Palo Alto Firewall filter arguments must specify from-zone and to-zone.')
+
+            self.from_zone = filter_options[1]
+            self.to_zone = filter_options[3]
+
+            if len(filter_options) > 4:
+                filter_type = filter_options[4]
+            else:
+                filter_type = 'inet'
+
+            if filter_type not in self._SUPPORTED_AF:
+                raise UnsupportedHeader(
+                    'Palo Alto Firewall Generator currently does not support %s as a header option' %
+                    (filter_type))
+
+            term_dup_check = set()
+            new_terms = []
+            for term in terms:
+                term.name = self.FixTermLength(term.name)
```

Comment:
There is a variable named [_TERM_MAX_LENGTH](https://github.com/google/capirca/blob/master/lib/aclgenerator.py#L265). If there is a max length FixTermLength should be called and _TERM_MAX_LENGTH should be set. This will raise an error if the term name is to long. If there is no limit then FixTermLength is irrelevant. Make sense?